src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java

```diff
@@ -1070,17 +1070,9 @@ public ResolvedWorkspaceID resolveWorkspace(
 	@Override
 	public Map<ResolvedWorkspaceID, Map<User, Permission>> getAllPermissions(
 			final Set<ResolvedWorkspaceID> rwsis)
-			throws WorkspaceCommunicationException,
-			CorruptWorkspaceDBException {
-		final Map<ResolvedWorkspaceID, Map<User, Permission>> res =
-				query.queryPermissions(rwsis, null);
-		final Map<ResolvedWorkspaceID, Map<User, Permission>> ret = 
-				new HashMap<ResolvedWorkspaceID, Map<User, Permission>>();
-		//probably a better way to do this
-		for (final ResolvedWorkspaceID r: res.keySet()) {
-			ret.put((ResolvedWorkspaceID)r, res.get(r)); 
-		}
```

Comment:
This is bizarre - lines 1077 on are a no-op. I think this is cruft from a completely crazy initial design that subclassed the resolved WS ID somehow, but I forget the details.